protobufs/livekit_models.proto

```diff
@@ -601,3 +601,60 @@ enum AudioTrackFeature {
   TF_ENHANCED_NOISE_CANCELLATION = 5;
 }
 
+message DataStream {
+
+  // enum for operation types (specific to TextStreamHeader)
+  enum OperationType {
+    CREATE = 0;
+    UPDATE = 1;
+    DELETE = 2;
+    REACTION = 3;
+  }
+  
+  // header properties specific to text streams
+  message TextHeader {
+    OperationType operationType = 1;
+    int32 version = 2;  // Optional: Version for updates/edits
+    string replyToStreamId = 3;  // Optional: Reply to specific message
+    repeated string attachedStreamIds = 4;
+    bool generated = 5; // true if the text has been generated by an agent from a participant's audio transcription
+
+  }
+  
+  // header properties specific to file or image streams
+  message FileHeader {
+    string fileName = 1;  // optional: name of the file
+  }
+
+  // main DataStreamHeader that contains a oneof for specific headers
+  message Header {
+    string streamId = 1; // unique identifier for this data stream
+    int64 timestamp = 2;  // using int64 for Unix timestamp
+    string topic = 3;
+    string mimeType = 4;
+    optional uint64 totalLength = 5;  // only populated for finite streams, if it's a stream of unknown size this stays empty
+    optional uint64 totalChunks = 6;  // only populated for finite streams, if it's a stream of unknown size this stays empty
+    Encryption.Type encryptionType = 7;  // defaults to NONE
+    map<string, string> extensions = 8;  // user defined extensions map that can carry additional info
+  
+    // oneof to choose between specific header types
+    oneof content_header {
+      TextHeader textHeader = 9;
+      FileHeader fileHeader = 10;
+    }
+  }
+
+  message Packet {
+    string messageId = 1;  // unique identifier for this data stream to map it to the correct header
+    uint64 chunkIndex = 2; // chunkId as an unsigned 32-bit integer
+    bytes content = 3; // content as binary (bytes)
+    uint32 contentLength = 4; // content length in bytes
```

Comment:
what is the purpose of this? can this not be derived from `content` as length of it? Are there cases where content is padded?  (maybe for the encryption case?)

Reply:
yeah, although I'm not sure anymore what we'd need it for, even in the encryption case. Will remove and we can re-add if it turns out we need it